account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/SendOtpNotificationHandler.java

```diff
@@ -246,20 +252,68 @@ public APIGatewayProxyResponseEvent sendOtpRequestHandler(
                             userLanguage);
                 case VERIFY_PHONE_NUMBER:
                     LOG.info("NotificationType is VERIFY_PHONE_NUMBER");
-                    var existingPhoneNumber =
-                            dynamoService
-                                    .getUserProfileByEmailMaybe(email)
-                                    .map(UserProfile::getPhoneNumber)
-                                    .orElse(null);
-                    var phoneNumberValidationError =
-                            ValidationHelper.validatePhoneNumber(
-                                    existingPhoneNumber,
-                                    sendNotificationRequest.getPhoneNumber(),
-                                    configurationService.getEnvironment());
-                    if (phoneNumberValidationError.isPresent()) {
-                        return generateApiGatewayProxyErrorResponse(
-                                400, phoneNumberValidationError.get());
+
+                    var userProfile = dynamoService.getUserProfileByEmailMaybe(email);
+
+                    if (userProfile.isEmpty()) {
+                        LOG.error(NO_USER_PROFILE_FOR_EMAIL.getMessage());
+                        return generateApiGatewayProxyErrorResponse(400, NO_USER_PROFILE_FOR_EMAIL);
+                    }
+
+                    String newPhoneNumber;
+                    try {
+                        newPhoneNumber =
+                                PhoneNumberHelper.formatPhoneNumber(
+                                        sendNotificationRequest.getPhoneNumber());
+                    } catch (Exception e) {
+                        return generateApiGatewayProxyErrorResponse(400, INVALID_PHONE_NUMBER);
                     }
+
+                    if (!userProfile.get().getMfaMethodsMigrated()) {
```

Comment:
Instead of conditioning here, can we use the MFAMethodService here to get the methods so we don't need to care about migration, which I think would simplify the implementation here.

FWIW I've started adding a helper utility for finding items in the List<MFAMethod> in an open PR, but this is probably not exactly what you need here https://github.com/govuk-one-login/authentication-api/blob/AUT-4248/use-backup-mfa/shared/src/main/java/uk/gov/di/authentication/shared/services/mfa/MFAMethodsService.java#L231C1-L249C6